### ISSUE TITLE
RC-V0.1.0 - BASE BRANCH

### ISSUE THREAD

**Issue by nathanielBellamy** [nathanielBellamy/my_website#34]
# [PR](https://github.com/nathanielBellamy/my_website/pull/35)

This is the base branch for our next planned release.

# Branches

### ✅0. `MW-32-go-backend-ii` WebSocket Shenanigans
  - what is effectively the base branch at the moment

=========UPDATE=========
- has been merged into new base branch

=======END UPDATE=======

### ✅a. persist wasm through resize (generally handle resize better)

=========UPDATE=========
 - 📣📣📣
 - this one got handled leading to much simplification
 - resize handled from within wasm
 - retrofitted MagicSquare, so it can enjoy the simplifications and better-ness of it
 - the missing sauce was updating `WebGL2RenderingContext.viewport(x,y,w,h)`
 - handled in `MW-32-go-bakend-ii`
 
=======END UPDATE=======

  -  it was passable to reload it every resize when all we had to do was re-instantiate the wasm module
  - however, now that wasm is connecting to a websocket, which we don't want to be continuously building and destroying
  - this should be doable by passing a js closure into `.run()` as we did for setting all settings
  - this is actually how `MagicSquare` should work fundamentally, so changes here may more deeply impact `MagicSquare`
  - it's possible, delaying initial render until websocket has returned initial settings from server is closely related to the changes here - if so, do that while here
  
 ###✅ b. add an explanation screen when signing on to `PublicSquare`
  - don't worry about local storage
  - this could be used to hide the initial render of outdated/default settings

### ✅c. fix connected toast auto disappear
  - currently it is not

#### POSTPONED [ticket](https://github.com/users/nathanielBellamy/projects/4/views/1?pane=issue&itemId=38112162)
 - have Go write the settings to file every 10 minutes
 - impl [gocron](https://pkg.go.dev/github.com/go-co-op/gocron)
 - have Go read from the file on server start
 - if no file found, create it with default before starting the pool

### ✅e. create a nice system diagram
  - make it publicly available even if code isn't

#### POSTPONED [ticket](https://github.com/users/nathanielBellamy/projects/4/views/1?pane=issue&itemId=38112209)
  - create nateschieberdevstatus@gmail.com
  - impl [go-smtp](https://github.com/emersion/go-smtp)
  - send a site status email every hour
  - determine status with a basic cypress test against prod

### ✅g. MS Mobile Layout
 - on small screens, allow only viewing either MS xor Controls
 - ensure that no scrolling happens so that touch tracking is allowed to work fully
 - increase pull-down reload threshold if possible
 - either makes sure this fixes multiple color mods bug or create a ticket to fix it

### ✅h. Set up a dev server
- require login 
- using nginx to forceSSL

###  ✅k.Fix LFO rate doubling on refresh (might be fixed by adequately handling resize)
=========UPDATE=========
 - 📣📣📣
 - this was indeed fixed by handling resize appropriately
 - handled in `MW-32-go-bakend-ii`

=======END UPDATE=======

### ✅m. Finish feed
- emojis only
- styling
- hide in MagicSquare, show in PublicSquare

#### POSTPONED [ticket](https://github.com/users/nathanielBellamy/projects/4/views/1?pane=issue&itemId=38112318)
- this may be a separate, later release
- in the event the data coming from the server is corrupted
- clients should send a message to the server acknowledging the event, meanwhile setting default values in the client
- once the server receives a failure message it should pause all updates until it ensures all clients agree on the default settings

#### POSTPONED [ticket](https://github.com/users/nathanielBellamy/projects/4/views/1?pane=issue&itemId=38112347)
  - in both squares
  - Flowbite Drawer with some helpful tips

### ✅s. Debug - PublicSquare -> MagicSquare
  - MS renders and then is removed when transitioning from PS-> MS
  - fixed by refresh
  - all other siteSection transitions work fine, including MS -> PS
  - this was handled by adding loading screens
  - it seems to be an issue with how `svelte-router` handles the loading and unloading of components

### 🟦t. Improve messaging over connection error to PS
  - make style change (MS canvas container fade color) depending on connection status
## ✅u. pass Vite MODE into Wasm so Wasm knows where it is running
  - at least pass in the URL so Wasm knows where to open the websocket
  - or maybe we can access it directly using https://doc.rust-lang.org/std/macro.env.html

##  ✅v. Set up remote env and Recaptcha
  - continuation of `small-screen` branch

## ✅w. System diagram site section
  - link to the Repo
  - display diagram

## ✅x. Update ReadMe
  - document new npm scripts

## ✅z. Low-tech password protect dev site
